plugin/externalPluginEntry.go

```diff
@@ -141,12 +187,101 @@ func (e *externalPluginEntry) ChildSchemas() []*EntrySchema {
 	return []*EntrySchema{}
 }
 
+const schemaFormat = `{
+	"type_id_one":{
+		"label": "one",
+		"methods": ["list"],
+		"children": ["type_id_two"]
+	},
+	"type_id_two":{
+		"label":"two",
+		"methods": ["read"]
+	}
+}
+`
+
 func (e *externalPluginEntry) Schema() *EntrySchema {
-	// TODO: Support external plugin schemas.
-	return nil
+	if !e.implements("schema") {
+		return nil
+	}
+	var graph *linkedhashmap.Map
+	if e.schemaGraphs != nil {
+		g, ok := e.schemaGraphs[e.typeID]
+		if !ok {
+			msg := fmt.Errorf(
+				"e.Schema(): entry schemas were prefetched, but no schema graph was provided for %v (%v)",
+				ID(e),
+				e.typeID,
+			)
+			panic(msg)
+		}
+		graph = g
+		// As a sanity check, ensure that the methods specified in the entry's schema
+		// match the methods specified in the entry instance. Log a warning if there
+		// is a mismatch. Hopefully this should never happen.
+		es, _ := graph.Get(e.typeID)
+		schemaMethods := es.(entrySchema).Actions
+		instanceMethods := e.supportedMethods()
+		sort.Strings(schemaMethods)
+		sort.Strings(instanceMethods)
+		mismatchMsg := fmt.Sprintf(
+			"%v (%v): the schema's supported methods (%v) don't match the instance's supported methods (%v)",
+			ID(e),
+			e.typeID,
+			strings.Join(schemaMethods, ", "),
+			strings.Join(instanceMethods, ", "),
+		)
+		if len(schemaMethods) != len(instanceMethods) {
+			log.Warnf(mismatchMsg)
+		} else {
+			for i := range instanceMethods {
+				if instanceMethods[i] != schemaMethods[i] {
+					log.Warnf(mismatchMsg)
+					break
+				}
+			}
+		}
+	} else {
+		// Entry schemas were not prefetched, so we'll need to shell out. Even though entry
```

Comment:
In the gem, it would look something like
```
Wash.exclude_schema
```
so the code would know not to include the schema in root.